File: Practica1/Backend-python/app/controllers/movie_controller.py
from flask import Blueprint, jsonify
from app.services.movie_service import list_movies
import os

movie_bp = Blueprint("movie_bp", __name__)

@movie_bp.route("/", methods=["GET"])
def get_movies():
    try:
        movies = list_movies()
        bucket = os.getenv("AWS_BUCKET_NAME")
        region = os.getenv("AWS_REGION")

        base_url = f"https://{bucket}.s3.{region}.amazonaws.com"

        data = []
        for m in movies:
            data.append({
                "id_pelicula": m["id_pelicula"],
                "titulo": m["titulo"],
                "director": m["director"],
                "anio_estreno": m["anio_estreno"],
                "poster_url": f"{base_url}/{m['poster']}",
                "url_contenido": m["url_contenido"],
                "estado": m["estado"],
                "is_available": m["estado"] == "Disponible"
            })

        return jsonify({
            "success": True,
            "total": len(data),
            "data": data
        })

    except Exception as e:
        return jsonify({"success": False, "message": str(e)}), 500

File: Practica1/Backend-Nodejs/src/controllers/movie.controller.js
const movieService = require("../services/movie.service");
const { formatTimeAgo } = require("../utils/time");

async function getMovies(req, res, next) {
  try {
    const movies = await movieService.listMovies();

    const bucket = process.env.AWS_BUCKET_NAME;
    const region = process.env.AWS_REGION;

    const baseUrl = `https://${bucket}.s3.${region}.amazonaws.com`;

    const data = movies.map(m => ({
      id_pelicula: m.id_pelicula,
      titulo: m.titulo,
      director: m.director,
      anio_estreno: m.anio_estreno,
      poster_url: `${baseUrl}/${m.poster}`,
      url_contenido: m.url_contenido,
      estado: m.estado,
      is_available: m.estado === "Disponible"
    }));

    res.json({
      success: true,
      total: data.length,
      data
    });
  } catch (error) {
    next(error);
  }
}

module.exports = { getMovies };